irony-cdb-json.el

```diff
@@ -111,17 +161,19 @@ directories to project directory."
            finally return found))
 
 (defun irony-cdb-json--locate-db ()
```

Comment:
1. Do I understand correctly that the `irony-cdb-json--project-alist` is no longer used?
2. What about people who put their build directory in `Build`, or maybe, `build/release`, `build/Debug`, `build-clang`, `build-gcc-x86_64`, ...? I don't really want to hardcode `build` anywhere in irony, it should just be easy for the user to customize the lookup with his preferred builds.
3. please favor `locate-dominating-file` over custom logic. To understand why, see `locate-dominating-stop-dir-regexp`, `locate-dominating-file`, `expand-file-name`, ...